### PR TITLE
testing guide updates

### PR DESCRIPTION
**Pull request by toddjordan**
Part of guides review for https://github.com/emberjs/guides/issues/602

I've implemented the guides in an ember cli project: https://github.com/toddjordan/ember-testing-guide-examples to ensure they work on ember 2.1 beta, and I've also pulled in ember suave to check style concerns. 

In implementing the examples I've made updates as I go to make sure the instructions are clear and that the examples work.  I've also added some unit test style things like naming tests with 'should...' so that they show up readable and behavior focused in the output, as well as adding assert descriptions, so that the asserts when printed in out put are readable.

Still working on testing intro page.. I'd like to add in the concept of integration test there and more appropriately split out the examples.  

The component guide page is part of issue https://github.com/emberjs/guides/issues/501 and pull request https://github.com/emberjs/guides/pull/774

### CODE REVIEW

File: source/testing/testing-models.md [hunk 1]
```diff
@@ -48,9 +55,11 @@ declarations are setup properly.
 
 Assume that a `User` can own a `Profile`.
 
+> You can follow along by generating your own user and profile models with `ember
+> generate model user` and `ember generate model profile`
```
Comment (michaelrkn) [inline source/testing/testing-models.md]: Should have a period at the end of the sentence.

File: source/testing/unit-testing-basics.md [hunk 2]
```diff
@@ -11,39 +11,43 @@ common cases.
...
+Let's start by creating an object that has a `computedFoo` computed property
 based on a `foo` property.
 
 ```app/models/some-thing.js
 export default Ember.Object.extend({
   foo: 'bar',
 
-  computedFoo: Ember.computed('foo', function() {
-    return 'computed ' + this.get('foo');
+computedFoo: Ember.computed('foo', function() {
```
Comment (michaelrkn) [inline source/testing/unit-testing-basics.md]: indent

**Comment by michaelrkn** [comment]
Looks great to me! I added a couple notes. @locks can you review as well?

**Comment by locks** [comment]
:+1:

**Pull request merged**